src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -478,6 +476,51 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
         });
   }
 
+  /**
+   * Redirects to login page if feedback session is accessible.
+   *
+   * @param auth The authentication information used in the redirection URL.
+   */
+  redirectToLoginPageIfFeedbackSessionAccessible(auth: AuthInfo): void {
+    this.loadCourseInfo();
+    this.loadPersonName();
+    this.isFeedbackSessionLoading = true;
+    this.feedbackSessionsService.getFeedbackSession({
+      courseId: this.courseId,
+      feedbackSessionName: this.feedbackSessionName,
+      intent: this.intent,
+      key: this.regKey,
+      moderatedPerson: this.moderatedPerson,
+      previewAs: this.previewAsPerson,
+    }).pipe(finalize(() => {
+      this.isFeedbackSessionLoading = false;
+    }))
+      .subscribe(() => {}, (resp: ErrorMessageOutput) => {
+        if (resp.status === 404) {
```

Comment:
If we take out the additional redirect in from the `resp.status === 403 && resp.error.message === 'Login is required to access this feedback session'`, this entire block is a carbon copy of the error handling in the other method. See if you can find a way to extract them out.